x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/support/SpnegoClient.java

```diff
@@ -0,0 +1,253 @@
+        loginContext.login();
+        return loginContext;
+    }
+
+    /**
+     * Usually we would have a JAAS configuration file for login configuration.
+     * Instead of an additional file setting as we do not want the options to be
+     * customizable we are constructing it in memory.
+     * <p>
+     * As we are uing this instead of jaas.conf, this requires refresh of
+     * {@link Configuration} and reqires appropriate security permissions to do so.
+     */
+    static class PasswordJaasConf extends Configuration {
+        private final String principal;
+
+        PasswordJaasConf(final String principal) {
+            this.principal = principal;
+        }
+
+        @Override
+        public AppConfigurationEntry[] getAppConfigurationEntry(final String name) {
+            final Map<String, String> options = new HashMap<>();
+            options.put("principal", principal);
+            options.put("storeKey", Boolean.TRUE.toString());
+            options.put("useTicketCache", Boolean.FALSE.toString());
+            options.put("useKeyTab", Boolean.FALSE.toString());
+            options.put("renewTGT", Boolean.FALSE.toString());
+            options.put("refreshKrb5Config", Boolean.TRUE.toString());
+            options.put("isInitiator", Boolean.TRUE.toString());
+            options.put("debug", Boolean.TRUE.toString());
+
+            return new AppConfigurationEntry[] { new AppConfigurationEntry(SUN_KRB5_LOGIN_MODULE,
+                    AppConfigurationEntry.LoginModuleControlFlag.REQUIRED, Collections.unmodifiableMap(options)) };
+        }
+    }
+
+    /**
+     * Jaas call back handler to provide credentials.
+     */
+    static class KrbCallbackHandler implements CallbackHandler {
+        private final String principal;
+        private final SecureString password;
+
+        KrbCallbackHandler(final String principal, final SecureString password) {
+            this.principal = principal;
+            this.password = password;
+        }
+
+        public void handle(final Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+            for (Callback callback : callbacks) {
+                if (callback instanceof PasswordCallback) {
+                    PasswordCallback pc = (PasswordCallback) callback;
+                    if (pc.getPrompt().contains(principal)) {
+                        pc.setPassword(password.getChars());
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private static String getAndSetSystemProperty(final String systemProperty, final String value) {
+        String retVal = null;
+        try {
+            retVal = AccessController.doPrivileged(new PrivilegedExceptionAction<String>() {
+
+                @Override
+                @SuppressForbidden(
+                        reason = "For testing application provides credentials, needs sys prop javax.security.auth.useSubjectCredsOnly")
```

Comment:
This statement is pretty vague. Is this method really only used for a single system property?